eureka-client/src/main/java/com/netflix/discovery/provider/DiscoveryJerseyProvider.java

```diff
@@ -263,4 +218,33 @@ private static ISerializer getSerializer(@SuppressWarnings("rawtypes") Class ser
         }
         return converter;
     }
+
+    private static Response createErrorReply(int status, Throwable cause, MediaType mediaType) {
+        StringBuilder sb = new StringBuilder(cause.getClass().getName());
+        if (cause.getMessage() != null) {
+            sb.append(": ").append(cause.getMessage());
+        }
+        return createErrorReply(status, sb.toString(), mediaType);
+    }
+
+    private static Response createErrorReply(int status, String errorMessage, MediaType mediaType) {
+        String message;
+        if (MediaType.APPLICATION_JSON_TYPE.equals(mediaType)) {
+            message = "{\"error\": \"" + errorMessage + "\"}";
+        } else {
+            message = "<error><message>" + errorMessage + "</message></error>";
+        }
+        return Response.status(status).entity(message).type(mediaType).build();

```

Comment:
We should verify media type in isReadable method, but there is no proper check there. I will add it.